### PR TITLE
[feat]: [CDS-45294]: Added spot connector

### PR DESCRIPTION
**Pull request by arvind-choudhary-h**
We have added spot connector.
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

**Comment by rathodmeetsatish** [comment]
Please add a .txt file changelog folder correponding to your pr number describing your changes

File: docs/data-sources/platform_connector_spot.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_connector_spot Data Source - terraform-provider-harness"
+subcategory: ""
```
Comment (rathodmeetsatish) [inline docs/data-sources/platform_connector_spot.md:4]: manually add subcategory as Next Gen

File: docs/resources/platform_connector_spot.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_connector_spot Resource - terraform-provider-harness"
+subcategory: ""
```
Comment (rathodmeetsatish) [inline docs/resources/platform_connector_spot.md:4]: same as above

File: docs/data-sources/platform_connector_spot.md (hunk 1 above)
Comment (arvind-choudhary-h) [reply docs/data-sources/platform_connector_spot.md]: Done

File: docs/resources/platform_connector_spot.md (hunk 2 above)
Comment (arvind-choudhary-h) [reply docs/resources/platform_connector_spot.md]: done

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**